apps/examples/speech_detector_test/speech_detector_test_main.cpp

```diff
@@ -0,0 +1,122 @@
+/****************************************************************************
+ *
+ * Copyright 2018 Samsung Electronics All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied. See the License for the specific
+ * language governing permissions and limitations under the License.
+ *
+ ****************************************************************************/
+
+/****************************************************************************
+ * Included Files
+ ****************************************************************************/
+
+#include <tinyara/config.h>
+#include <stdio.h>
+#include <apps/platform/cxxinitialize.h>
+
+#include <media/MediaRecorder.h>
+#include <media/MediaRecorderObserverInterface.h>
+#include <media/BufferOutputDataSource.h>
+
+#include <media/voice/SpeechDetectorInterface.h>
+
+/****************************************************************************
+ * speech_detector_test_main
+ ****************************************************************************/
+
+static void run() {
```

Comment:
We don't need recorder in app level.
KD should do that by itself
sw kd -> including recorder and do it in loop
hw kd -> init & wait until interrupt occur